CorsixTH/Lua/audio.lua

```diff
@@ -587,11 +587,24 @@ function Audio:playBackgroundTrack(index)
       -- Someone might want to stop the player from
       -- starting to play once it's loaded though.
       self.load_music = true
-      SDL.audio.loadMusicAsync(data, function(music_data, e)
+      SDL.audio.loadMusicAsync(data, function(music_data, err)
 
         if music_data == nil then
-          error("Could not load music file \'" .. (info.filename_music or info.filename) .. "\'" ..
-              (e and " (" .. e .. ")" or ""))
+          info.enabled = false
+          local name, msg = (info.filename_music or info.filename)
+          if not self.warned then -- Warn once per session
+            self.app.ui:addWindow(UIInformation(self.app.ui, {_S.errors.music}))
+          end
+          self.warned = true
+          if err == "No SoundFonts have been requested" then
+            msg = "Required soundfont is not found, please download one. A suitable soundfont is linked from the CorsixTH wiki."
+          elseif err == "XMP: Unrecognized file format" or err == "ModPlug_Load failed" then
+            msg = "Music format not supported for file " .. name
```

Comment:
Are these interpretations always true?

Reply:
Don't know. The catch-all way they are used in SDL Mixer means we can't tell some error situations apart.

Reply:
In light of that it could be too much to infer the error is a single truth. I'm not precious and happy to keep it as is though.
Does `name` contain the file extension?

Reply:
It could give the original error, plus "This usually means" and these new messages. Or instead of this change, add the details to the wiki.  
Yes, full path.

Reply:
I don't think it's worth the confusion on reflection. Full path is good enough.